pkg/manager/manager.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manager
+
+import (
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// TODO potentially this can be extracted from _some_
+// configuration (i.e. crossplane itself and providers).
+// New attributes can be added to set enable/disable this
+// and also the number of desired replicas, with some sane
+// default for production deployment. This can be useful to
+// opt out of this 1) on development 2) CI tests 3) per user
+// decision.
+var leaderElection = true
+
+// NewManager returns a new Manager for creating Controllers.
+// This takes crossplane specific configuration into consideration
+// to either enable leader election for the said Manager or opt
+// out of it.
+func NewManager(config *rest.Config, options ctrl.Options) (ctrl.Manager, error) {
+	if leaderElection {
+		options.LeaderElection = true
+		options.LeaderElectionID = "crossplane-leader-election-id"
```

Comment:
That's true, I was only demonstrating the idea for how this new change could look like. For example, we need to pass down more information and metadata about the controller-manager to `NewManager` function.

One option could be that we replace `ctrl.Options` with `ctrlmanager.Option` as well which will be crossplane specific set of options (name of manager, version of the manager, should leader election ever be enabled for this manager or not and etc).

I mean [theses](https://github.com/crossplane/crossplane-runtime/pull/220/files#diff-d500fbd6a2aa620607ca5e2a7c3ac4f1a4c82309d1a549561e92abfcb18f2f0eR24-R31) have to come from _somewhere_.

Reply:
@khos2ow exposing a different set of options is an interesting idea! I had iniitally thought about setting defaults in the `ctrl.Options` if they weren't already defined, but that was a bit scary because you could unexpectedly have a value set that conflicted with what you defined. I am still trying to determine if abstracting this at all is worth it given that this is likely not a value that folks will frequently modify. For instance, if we just configure it correctly in https://github.com/crossplane/provider-template maybe that is enough?

Reply:
I think you're right, these won't change that much but I think some sort of control over the values (some of them at least) makes sense. Because in reality leader election only makes sense where there is more than one replica of the manager and that IMO should be controllable (easily) thorough YAML (`ProviderConfig` maybe? I'm still trying to understand what's where :sweat_smile:), but for example the election algorithm should be baked into the provider and I believe that can be added in `provider-template` as an opinionated default.